src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -0,0 +1,522 @@
+            final File preprocessedFile = new File(workingDir, preppedFileName);
+
+            preppedFileName = runAndStopEarly("correct", preprocessedFile, sgaPath, workingDir, threadsToUse, indexer, indexerArgs, runtimeInfo);
+            if( null == preppedFileName ){
+                return new SGAAssemblyResult(null, runtimeInfo);
+            }
+            final File correctedFile = new File(workingDir, preppedFileName);
+
+            preppedFileName = runAndStopEarly("filter", correctedFile, sgaPath, workingDir, threadsToUse, indexer, indexerArgs, runtimeInfo);
+            if( null == preppedFileName ){
+                return new SGAAssemblyResult(null, runtimeInfo);
+            }
+            final File filterPassingFile = new File(workingDir, preppedFileName);
+
+            preppedFileName = runAndStopEarly("rmdup", filterPassingFile, sgaPath, workingDir, threadsToUse, indexer, indexerArgs, runtimeInfo);
+            if( null == preppedFileName ){
+                return new SGAAssemblyResult(null, runtimeInfo);
+            }
+        }
+
+        final File fileToMerge      = new File(workingDir, preppedFileName);
+        final String fileNameToAssemble = runAndStopEarly("fm-merge", fileToMerge, sgaPath, workingDir, threadsToUse, indexer, indexerArgs, runtimeInfo);
+        if(null == fileNameToAssemble){
+            return new SGAAssemblyResult(null, runtimeInfo);
+        }
+
+        final File fileToAssemble   = new File(workingDir, fileNameToAssemble);
+        final String contigsFileName = runAndStopEarly("assemble", fileToAssemble, sgaPath, workingDir, threadsToUse, indexer, indexerArgs, runtimeInfo);
+        if(null == contigsFileName){
+            return new SGAAssemblyResult(null, runtimeInfo);
+        }
+
+        // if code reaches here, all steps in the SGA pipeline went smoothly, but the following conversion from File to ContigsCollection may throw IOException
+        final File assembledContigsFile = new File(workingDir, contigsFileName);
+        return new SGAAssemblyResult(assembledContigsFile, runtimeInfo);
+    }
+
+    /**
+     * Call the right sga module, log runtime information, and return the output file name if succeed.
+     * If process erred, the string returned is null.
+     * @param moduleName            SGA module name to be run
+     * @param inputFASTQFile        FASTQ file tobe fed to SGA modul
+     * @param sgaPath               full path to the SGA program
+     * @param workingDir            directory the SGA pipeline is working in
+     * @param threads               threads to use
+     * @param indexer               module representing SGA index
+     * @param indexerArgs           arguments used by SGA index
+     * @param collectiveRuntimeInfo runtime information collected along the process
+     * @return                      the name of file produced by running this SGA module
+     */
+    private static String runAndStopEarly(final String moduleName,
```

Comment:
The `GATKException` was a bad decision, but I couldn't think of a better one.
Essentially, this is to make sure that no non-sense sga modules like `sga destroy_everything` is called.
Any suggestions on how to do that?
On the other hand, the name `runAndStopEarly` is to indicate to its caller, `runSGAModulesInSerial`, that something might be wrong in this so it this module call erred, make an early return.
Admittedly, bad name.